src/router.ts

```diff
@@ -394,35 +411,66 @@ export async function handleNewVoter(
     userInfo.isDemo
   );
 
-  let slackChannelName = userInfo.isDemo ? 'demo-lobby' : 'lobby';
-  if (entryPoint === LoadBalancer.PULL_ENTRY_POINT) {
-    logger.debug(
-      `ROUTER.handleNewVoter (${userInfo.userId}): New voter will enter Slack channel: ${slackChannelName}`
-    );
-  } else if (entryPoint === LoadBalancer.PUSH_ENTRY_POINT) {
-    userInfo.stateName = LoadBalancer.getPushPhoneNumberState(
-      twilioPhoneNumber
-    );
-    logger.debug(
-      `ROUTER.handleNewVoter (${userInfo.userId}): Determined that twilioPhoneNumber ${twilioPhoneNumber} corresponds to U.S. state ${userInfo.stateName} based on hard coding in LoadBalancer.`
-    );
-    const selectedChannelName = await LoadBalancer.selectSlackChannel(
-      redisClient,
-      LoadBalancer.PUSH_ENTRY_POINT,
-      userInfo.stateName
-    );
+  let slackChannelName = null as string | null;
 
-    logger.debug(
-      `ROUTER.handleNewVoter (${userInfo.userId}): LoadBalancer returned Slack channel ${selectedChannelName} for new PUSH voter.`
-    );
-    if (selectedChannelName) {
-      slackChannelName = selectedChannelName;
-    } else {
-      // If LoadBalancer didn't find a Slack channel, then select #national or #demo-national as fallback.
-      slackChannelName = userInfo.isDemo ? 'demo-national' : 'national';
-      logger.error(
-        `ROUTER.handleNewVoter (${userInfo.userId}): ERROR LoadBalancer did not find a Slack channel for new PUSH voter. Using ${slackChannelName} as fallback.`
+  // do we already know the voter's state?
+  const knownState = await DbApiUtil.getKnownPhoneState(
+    userOptions.userPhoneNumber
+  );
+  if (knownState) {
```

Comment:
If there is a knownState why do we call determineState?

Reply:
knownState will be ``MI`` but helpline wants ``Michigan``.

Reply:
Can we change the database table to state names so we don't have to do this parsing online?

Reply:
We could.. I kind of like it this way though because it adds a layer of protection against bad/bogus data in the table.

Reply:
Ok no problem whichever you prefer SGTM